[Raveen Prabhu] iP

itsraveen opened the pull request:
# DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* FAST SUPER FAST to use

All you need to do is,

1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
------------------------------------------------------------------------------
If you Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

itsraveen committed:
Level-1-Greet-Echo-Exit

itsraveen committed:
Level-2-Add-List

itsraveen committed:
Level-3-Mark-as-Done

itsraveen committed:
Level-4-ToDos-Events-Deadlines

itsraveen committed:
Level-5-Handle-Errors

itsraveen committed:
Level-6-Delete

itsraveen committed:
Level-6-Delete

itsraveen committed:
Add-Automated-Text-UI-Testing

itsraveen committed:
Automated Text UI Testing

itsraveen committed:
Update Duke.java

itsraveen committed:
Level-7-Save

itsraveen committed:
Level-8-Date-and-Time

itsraveen committed:
Merge pull request #1 from itsraveen/branch-Level-7

Level-7-Save

itsraveen committed:
Resolved merge conflicts

itsraveen committed:
Deleted unnecessary files

itsraveen committed:
Deleted unnecessary files

itsraveen committed:
Modified to follow OOP Principles

itsraveen committed:
Modified code to follow OOP Principles

itsraveen committed:
made the code cleaner

itsraveen committed:
Added A-JUnit

itsraveen committed:
Added A-JUnit

itsraveen committed:
Added A-JUnit

itsraveen committed:
Added JavaDocs

itsraveen committed:
Modified code to follow Coding Standard

itsraveen committed:
Level 9 added

itsraveen committed:
Merge branch 'branch-A-CodingStandard'

itsraveen committed:
Merge branch 'branch-Level-9'

itsraveen committed:
Added JavaDoc for findTask method

itsraveen committed:
Renamed bot to Victoria

itsraveen committed:
Add gradle

itsraveen committed:
Merge branch 'add-gradle-support'

itsraveen committed:
Create GUI and interact with user

itsraveen committed:
Level-10

itsraveen committed:
Bot is working with the GUI

itsraveen committed:
Cleaned code

Gabau commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,160 @@
+
+    /**
+     * Command to find and display all tasks containing keyword 
+     * @param keyword word to search for
```
There should be an empty line before param right?

Linda124 replied:
Should there be a blank line after the first statement and before the param in the comment?

Gabau commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,57 @@
+     * @return the symbol that shows if an event is done or not
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```
Are the parentheses necessary here? I do not think it is needed for the coding standard.

Gabau commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,55 @@
+    private static String LINES = "--------------------------------------------------------------------------------------\n";
+    
+    public static void intro() {
+        String logo = " ____        _        \n"
```
Would it help with the legibility of the code if the logos and messages were to be made into static constants?

Gabau commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,160 @@
+     */
+    public String deadlineTask(String str) {
+        try {
+            int slashIndex = str.indexOf("/");
```
The split method of a String object could have helped.

Gabau commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,166 @@
+    }
+
+    /**
+     * 
```
Missing method description.

Gabau reviewed (commented):
LGTM. The code is clean and tidy and the names look appropriate. All the best!

Linda124 commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,57 @@
+    /**
+     * A Method to mark the task as completed
+     */
+    public void taskDone() {
```
Perhaps a better method name would be markDone as it sounds more like a verb?

Linda124 commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,160 @@
+     * @return A response from the bot to indicate a deadline task has been created
+     * and added to the task list
+     */
+    public String deadlineTask(String str) {
```
Perhaps a more intuitive method name here is createDeadlineTask.

Linda124 commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,160 @@
+     * @return A response from the bot to indicate a event has been created
+     * and added to the task list
+     */
+    public String eventsTask(String str) {
```
Could createEventsTask be a clearer method name?

Linda124 commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,160 @@
+     * @return A response from the bot to indicate a to-do task has been created
+     * and added to the task list
+     */
+    public String todoTask(String str) throws DukeException {
```
Maybe you could consider naming this method createTodoTask instead?

Linda124 reviewed (commented):
Overall, it looks pretty good! There aren't many changes to be made.

itsraveen committed:
Update build.gradle

Created a new JAR file

The previous JAR file was created before the inclusion of the JavaFX libraries.

This new JAR file has included these libraries as well.

itsraveen committed:
Added assert statements

itsraveen committed:
Merge pull request #4 from itsraveen/branch-A-Assertions

Added assert statements

itsraveen committed:
A-CodeQuality

itsraveen committed:
Merge pull request #6 from itsraveen/branch-A-CodeQuality

A-CodeQuality

itsraveen committed:
Add assertions

Add assertions to document important assumptions in the code.

Some assumptions need to hold true and using assertions can help to
validate this on the developer's end.

Added some assertions to verify whether our assumptions hold true
during runtime.

itsraveen committed:
Update files to match code standard

Currently, the Duke class consists of a number of unused methods.

Let's refactor it to ensure it matches the code standard.

itsraveen committed:
Modified the MainWindow.fxml and DialogBox.fxml

The old look was not that nice with text often overflowing.

Need to make sure that the text do not overflow to make reading it easier.

Dialog box made bigger, height and width wise.

Let's change how it looks to give the user a better user experience/

itsraveen committed:
Extracted reply messages into Message Class

itsraveen committed:
Update files to match code standard

itsraveen committed:
Added Priority to tasks

itsraveen committed:
Merge pull request #8 from itsraveen/branch-C-Priority

Added Priority to tasks

itsraveen committed:
Show Priority method implemented

itsraveen committed:
Fixed bugs in storage

itsraveen committed:
Fixed bugs to allow reading from duke.txt

itsraveen committed:
Made the GUI better

itsraveen committed:
Merge pull request #9 from itsraveen/branch-A-BetterGui

Made the GUI better

itsraveen committed:
Add better GUI

itsraveen committed:
Added exit message

itsraveen committed:
Update user guide

itsraveen committed:
Add organisation into packages

itsraveen committed:
Add Exception

itsraveen committed:
Add missing Javadoc

itsraveen committed:
Update README

itsraveen committed:
Update README

itsraveen committed:
Fix storage issues

itsraveen committed:
Fix storage issues

itsraveen committed:
Update Readme

itsraveen committed:
Update Readme

itsraveen committed:
Update Readme

itsraveen committed:
Update Readme
